### PR TITLE
ZIM-28 Test for RedoableOp. Refactor for easier testing.

### PR DESCRIPTION
**Pull request by cgeisser**

### CODE REVIEW

File: store/src/java/com/zimbra/cs/redolog/op/RedoableOp.java [hunk 1]
```diff
@@ -75,15 +75,18 @@
     private boolean mUnloggedReplay;  // true if redo of this op is not redo-logged
     RedoCommitCallback mCommitCallback;
 
-    protected RedoableOp(MailboxOperation op) {
+    protected RedoableOp(MailboxOperation op, RedoLogManager mgr) {
```
Comment (spoon16) [inline store/src/java/com/zimbra/cs/redolog/op/RedoableOp.java]: 👍

File: store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        subOp.abort();
+        replayAll();
+        op.addChainedOp(subOp);
+
+        op.log(false);
+        op.abort();
+    }
+
+    @Test
+    public void checkSubclasses() throws Exception {
```
Comment (spoon16) [inline store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java]: How was/is this used? I see it was `static void main` before. Was it used by some CLI tool or something?

File: store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        subOp.commit();
+        replayAll();
+        op.addChainedOp(subOp);
+
+        op.log(false);
+        op.commit();
+    }
+
+    @Test
+    public void chainedAbort() {
```
Comment (spoon16) [inline store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java]: no assertions?

File: store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        Assert.assertNotNull("getInputStream sets up internal vector.",
+                             op.mSerializedByteArrayVector);
+        Assert.assertEquals(46, out.available());
+        byte[] bytes = new byte[RedoableOp.REDO_MAGIC.length()];
+        out.read(bytes);
+        Assert.assertEquals(RedoableOp.REDO_MAGIC, new String(bytes));
+    }
+
+    @Test
+    public void chainedCommit() {
```
Comment (spoon16) [inline store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java]: no assertions?

File: store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        Assert.assertEquals(7, op.getTimestamp());
+        op.log(false);
+        op.setSerializedByteArray(new byte[3]);
+        op.commit();
+        Assert.assertNull("Commit clears byte array.",
+                          op.mSerializedByteArrayVector);
+    }
+
+    @Test
+    public void inactiveOp() {
```
Comment (spoon16) [inline store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java]: no assertions?

File: store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java (hunk 2 above)
Comment (cgeisser) [inline store/src/java-test/com/zimbra/cs/redolog/op/RedoableOpTest.java]: I'm not sure how it was used as a main()... But the concept is to make sure all the op types can be constructed by the factory, and that seems like a worthwhile test.

**Comment by cgeisser** [comment]
Added explicit calls to verify the mock where needed. I can't tell if EasyMock does this by default or not.

**Pull request merged**